src/Nethermind/Nethermind.Trie/PatriciaTree.cs

```diff
@@ -536,9 +534,7 @@ ref TraverseBranches(startNode, ref startingPath, traverseContext) :
                     TreePath startingPath = TreePath.Empty;
                     ResolveNode(RootRef, in traverseContext, in startingPath);
                     if (_logger.IsTrace) TraceNode(in traverseContext);
-                    return ref RootRef.IsBranch ?
-                        ref TraverseBranches(RootRef, ref startingPath, traverseContext) :
-                        ref TraverseNode(RootRef, traverseContext, ref startingPath);
+                    return ref TraverseNode(RootRef, traverseContext, ref startingPath);
```

Comment:
This changes the callstack by pushing the `TraverseBranches` one level deeper (it will be `TraverseNode`->`TraverseBranches` now. Not sure if it impacts the performance, but looks like something that could make the stack much higher.

Reply:
I think this was explicit as Branches are common, and node then does the same resolve so it cuts out the cost of additional method call

Reply:
Having that said, this refactorization might introduce some deeper stacks / more calls.

Reply:
Hmm, if this refactor affects performance, then we should move everything into one method since it's on the critical path in PatriciaTree

But intresting comment, thx @Scooletz

Reply:
I'd `run node` or even `compare` even though some comments sound smart. So many times my intuition was disproved by GH actions 😓